File: June-18/03-06-18/java/01-Binary-Swap.java

import java.io.*;


class TestClass {
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        String str1, str2;
        str1 = input.readLine();
        str2 = input.readLine();
        int n = str1.length(), cnt = 0;
        
        for(int i = 0; i < n; i++)
        {
            if(str1.charAt(i) != str2.charAt(i))
            {
                cnt += 1;
            }
        }
        
        if(cnt % 2 != 0)
            System.out.println("-1");
        else
            System.out.println(cnt/2);
        
        
    }
}

File: June-18/03-06-18/python/01-Binary-Swap.py
str1 = input()
str2 = input()
n = len(str1)
cnt = 0

for i in range(0, n):
    if(str1[i] != str2[i]):
        cnt += 1
if(cnt % 2 != 0):
    print("-1")
else:
    print(cnt//2)
